File: tmdb-core/src/main/java/dev/reuss/tmdb/common/image/MovieImages.java
package dev.reuss.tmdb.common.image;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Image metadata for a TMDB movie.
 *
 * @param id        TMDB movie id
 * @param backdrops backdrop images
 * @param logos     logo images
 * @param posters   poster images
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record MovieImages(
        int id,
        List<TmdbImage> backdrops,
        List<TmdbImage> logos,
        List<TmdbImage> posters
) implements PosterBackdropLogoImagesResponse, TmdbModel {

    public MovieImages {
        backdrops = TmdbCollections.list(backdrops);
        logos = TmdbCollections.list(logos);
        posters = TmdbCollections.list(posters);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/common/image/MovieImages.kt
package dev.reuss.tmdb.common.image

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Image metadata for a TMDB movie.
 *
 * @property id        TMDB movie id
 * @property backdrops backdrop images
 * @property logos     logo images
 * @property posters   poster images
 */
@JvmRecord
data class MovieImages(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("backdrops")
    override val backdrops: List<TmdbImage> = emptyList(),
    @all:JsonProperty("logos")
    override val logos: List<TmdbImage> = emptyList(),
    @all:JsonProperty("posters")
    override val posters: List<TmdbImage> = emptyList(),
) : PosterBackdropLogoImagesResponse,
    TmdbModel
